### ISSUE TITLE
unable to install

### ISSUE THREAD

**Issue by klocel2** [xelkano/redmine_xapian#100]
I use a Debian
Linux redmine 4.9.0-6-amd64 #1 SMP Debian 4.9.88-1+deb9u1 (2018-05-07) x86_64 GNU/Linux

with MariaDB
Server version: 10.1.37-MariaDB-0+deb9u1 Debian 9.6

My Redmine works without the plugin, but  I have this error message when I try to install the plugins :
root@redmine:/usr/share/redmine# bundle exec rake redmine:plugins:migrate RAILS_ENV="production"
rake aborted!
Bundler::GemRequireError: There was an error while trying to load the gem 'activerecord-tablefree'.
Gem Load Error is: Unsupported ActiveRecord version
Backtrace for gem load error is:
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:130:in `<module:SingletonMethods>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:74:in `<module:Tablefree>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:36:in `<module:ActiveRecord>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:10:in `<top (required)>'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:91:in `require'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:91:in `block (2 levels) in require'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:86:in `each'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:86:in `block in require'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:75:in `each'
/usr/lib/ruby/vendor_ruby/bundler/runtime.rb:75:in `require'
/usr/lib/ruby/vendor_ruby/bundler.rb:106:in `require'
/usr/share/redmine/config/application.rb:5:in `<top (required)>'
/usr/share/redmine/Rakefile:4:in `require'
/usr/share/redmine/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/rake_module.rb:29:in `load'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/rake_module.rb:29:in `load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:703:in `raw_load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:104:in `block in load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:103:in `load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:82:in `block in run'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:80:in `run'
/var/lib/gems/2.3.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:22:in `load'
/usr/local/bin/rake:22:in `<main>'
Bundler Error Backtrace:
/usr/share/redmine/config/application.rb:5:in `<top (required)>'
/usr/share/redmine/Rakefile:4:in `require'
/usr/share/redmine/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.3.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'

Caused by:
ActiveRecord::Tablefree::Unsupported: Unsupported ActiveRecord version
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:130:in `<module:SingletonMethods>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:74:in `<module:Tablefree>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:36:in `<module:ActiveRecord>'
/var/lib/gems/2.3.0/gems/activerecord-tablefree-3.1.0/lib/activerecord-tablefree.rb:10:in `<top (required)>'
/usr/share/redmine/config/application.rb:5:in `<top (required)>'
/usr/share/redmine/Rakefile:4:in `require'
/usr/share/redmine/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.3.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
(See full trace by running task with --trace)

**Comment by picman** [comment]
According to the following [site](https://rubygems.org/gems/activerecord-tablefree) `activerecord > 5.0` is required by _activerecord-tablefree_ gem.
Have you updated your gems (`bundle update`)?
Try to install _activerecord-tablefree_ gem manually.

**Comment by klocel2** [comment]
Thanks for your help.

An update do not solve the issue :
gem list | grep activerecord
activerecord (5.2.2, 4.2.11, 4.2.7.1)
activerecord-mysql2-adapter (0.0.3)
activerecord-tablefree (3.1.8, 3.1.0)

but : 
bundle install |grep activerecord
Using activerecord 4.2.11
Using activerecord-tablefree 3.1.0

The problem come from the Debian Gemfile ?

**Comment by picman** [comment]
What's your Redmine version?

**Comment by klocel2** [comment]
I use the lastest Debian package for Redmine, this is the 4.01. Redmine 4.x use gem rails 5.2.  The Gemfile of the package use  a 4.2 gem rails, it's look like a redmine 3.4 gemfile. 
II try to to change the gems rails version, but there are many dependancies on the file. I did not work yet :.
Bundler could not find compatible versions for gem "activemodel":
  In snapshot (Gemfile.lock):
    activemodel (= 5.2.2)

  In Gemfile:
    rails (~> 5.2.2) was resolved to 5.2.2, which depends on
      activestorage (= 5.2.2) was resolved to 5.2.2, which depends on
        activerecord (= 5.2.2) was resolved to 5.2.2, which depends on
          activemodel (= 5.2.2)

    rails (~> 5.2.2) was resolved to 5.2.2, which depends on
      activestorage (= 5.2.2) was resolved to 5.2.2, which depends on
        activerecord (= 5.2.2) was resolved to 5.2.2, which depends on
          activemodel (= 5.2.2)

    protected_attributes was resolved to 1.0.0, which depends on
      activemodel (< 5.0, >= 4.0.0.beta)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

**Comment by klocel2** [comment]
Running bundle update do not solve the problem, it's the same issue, the Gemfile force to use a rails gem 4.2 version (I can see that in the Gemfile.lock) . Redmine Work but not with the Xapian plugins. l download the Redmine package 4.0.2 from the Redmine website and copie this Gemfile in my Redmine root directory, I update also actionpack-xml_parser gem.

bundle update
bundle install | grep actionpack-xml_parser
Using actionpack-xml_parser 2.0.1

I have now another error :
 bundle exec rake redmine:plugins:migrate RAILS_ENV="production" --trace
rake aborted!
NameError: uninitialized constant ActionDispatch::ParamsParser
/usr/share/redmine/config/application.rb:55:in `<class:Application>'
/usr/share/redmine/config/application.rb:8:in `<module:RedmineApp>'
/usr/share/redmine/config/application.rb:7:in `<top (required)>'
/usr/share/redmine/Rakefile:4:in `require'
/usr/share/redmine/Rakefile:4:in `<top (required)>'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/rake_module.rb:29:in `load'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/rake_module.rb:29:in `load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:703:in `raw_load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:104:in `block in load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:103:in `load_rakefile'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:82:in `block in run'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:186:in `standard_exception_handling'
/var/lib/gems/2.3.0/gems/rake-12.3.2/lib/rake/application.rb:80:in `run'
/var/lib/gems/2.3.0/gems/rake-12.3.2/exe/rake:27:in `<top (required)>'
/usr/local/bin/rake:22:in `load'
/usr/local/bin/rake:22:in `<main>'

**Comment by picman** [comment]
In this case I'd recommend using the previous release v1.6.9.

**Comment by klocel2** [comment]
Ok I got it, the 4.0.1 package version is not the stable version, in fact I download the 3.4 ....

**Comment by Rachna-ramesh** [comment]
which Xapian version is compatible with Redmine 4.0 version ??

**Comment by picman** [comment]
The latest released v2.0.0.

**Comment by Rachna-ramesh** [comment]
xapian bindings get installed successfully, but it does not reflect on the plugins folder in bitnami Redmine v4.0.3. After that when we were checking production.log then I got the error "install a ruby xapian package or an alternative xapian bindings ". When we solved this error and restart all the services of bitnami again we got the same error. I followed the steps which u have mentioned before queries i.e "https://xapian.org/download".
